dgisim/src/agents.py

```diff
@@ -103,29 +103,25 @@ def choose_action(self, history: List[GameState], pid: GameState.Pid) -> PlayerA
 
                 # Consolidation Functions / Helpers
                 def try_card_action(card_class, condition, tmp_dices):
-                    if cards.contains(card_class) and condition():
-                        if active_character is not None:
-                            action = CardAction(
-                                card_class,
-                                CharacterTargetInstruction(
-                                    tmp_dices,
-                                    StaticTarget(
-                                        pid,
-                                        Zone.CHARACTER,
-                                        active_character.get_id()
-                                    )
-                                )
+                    assert active_character is not None
+                    
+                    if not cards.contains(card_class) and not condition():
+                        return None
+                    
+                    action = CardAction(
+                        card_class,
+                        CharacterSelectAction(
+                            tmp_dices,
```

Comment:
if tmp_dices is None, the action cannot be taken, so return None

Reply:
btw, please add type signatures to the functions you defined, so it can be statically checked by mypy and IDE